FirebaseDatabaseUI/FirebaseArray.m

```diff
@@ -123,7 +123,11 @@ - (void)initListeners {
         NSUInteger fromIndex = [self indexForKey:snapshot.key];
         [self.snapshots removeObjectAtIndex:fromIndex];
 
-        NSUInteger toIndex = [self indexForKey:previousChildKey] + 1;
+        NSUInteger toIndex = 0;
+        NSUInteger prevIndex = [self indexForKey:previousChildKey];
```

Comment:
We should be handling the case where the previous child key is `nil` as well here